### PR TITLE
Add Cypress test for search block sort on property

### LINKED ISSUE #6226: Add Cypress test for search block sort on property
Fix Pr: https://github.com/plone/volto/pull/6224

### PR DESCRIPTION
**Pull request by iRohitSingh**
Fixes #6226

### CODE REVIEW

**Comment by iRohitSingh** [comment]
Screenshot 2024-08-01 at 9 00 43 PM

**Review by nileshgulia1 — COMMENTED** [review]
LGTM. Also please add a changelog.

File: packages/volto/cypress/tests/core/blocks/blocks-search.js [hunk 1]
```diff
@@ -558,4 +558,102 @@ describe('Search Block Tests', () => {
     cy.get('#toolbar-save > .icon').click();
     cy.wait(500);
   });
+  it('Search block - test on select 1 sort on in listin criteria sorton', () => {
```
Comment (stevepiercy) [inline packages/volto/cypress/tests/core/blocks/blocks-search.js:561]:
```suggestion
  it('Search block - test on select 1 sort on in listing criteria sort on', () => {
```

File: packages/volto/cypress/tests/core/blocks/blocks-search.js [hunk 2]
```diff
@@ -558,4 +558,102 @@ describe('Search Block Tests', () => {
...
+    cy.get(
+      '#default-query-0-query .fields:first-of-type > .field .react-select__option',
+    )
+      .contains('Page')
+      .click();
+    cy.get(
+      '#select-listingblock-sort-on > .react-select__control > .react-select__value-container',
+    ).click();
+    cy.findByText('Order in folder').click();
+    // Add one soron options below
```
Comment (stevepiercy) [inline packages/volto/cypress/tests/core/blocks/blocks-search.js:641]:
```suggestion
    // Add one sort on option below
```

File: packages/volto/cypress/tests/core/blocks/blocks-search.js [hunk 3]
```diff
@@ -558,4 +558,102 @@ describe('Search Block Tests', () => {
...
+    cy.findByText('Effective date').click();
+    cy.get('.field-wrapper-showSortOn .wrapper .ui label').click();
+    //save page
+    cy.get('#toolbar-save').click();
+
+    // then we are able to see title and value
+    cy.get('span.sorted-label').should('have.text', 'Sorted onEffective date');
+    cy.get('span.sorted-label-value').should('have.text', 'Effective date');
+  });
+  it('Search block - test on Only one sortOn options below.', () => {
```
Comment (stevepiercy) [inline packages/volto/cypress/tests/core/blocks/blocks-search.js:594]:
```suggestion
  it('Search block - test on only one sort on option below.', () => {
```

File: packages/volto/cypress/tests/core/blocks/blocks-search.js [hunk 4]
```diff
@@ -558,4 +558,102 @@ describe('Search Block Tests', () => {
...
+    cy.get('#toolbar-save').click();
+    // then we are able to see label and sort option
+    cy.get('.sort-label').should('have.text', 'Sort on');
+    cy.get('#select-search-sort-on  ').click();
+    cy.findByText('Effective date').click({ force: true });
+    cy.get(
+      'div#select-search-sort-on.search-react-select-container.css-2b097c-container',
+    ).contains('Effective date');
+  });
+  it('Search block - test on Select both Listing sorton and Sorton options', () => {
```
Comment (stevepiercy) [inline packages/volto/cypress/tests/core/blocks/blocks-search.js:617]:
```suggestion
  it('Search block - test on select both listing sort on and sort on options', () => {
```

File: packages/volto/cypress/tests/core/blocks/blocks-search.js [hunk 5]
```diff
@@ -558,4 +558,102 @@ describe('Search Block Tests', () => {
...
+      .click();
+    cy.get(
+      '#select-listingblock-sort-on > .react-select__control > .react-select__value-container',
+    ).click();
+    cy.findByText('Order in folder').click();
+    // Add one soron options below
+    cy.get('.field-wrapper-showSortOn .wrapper .ui label').click();
+    cy.get('#field-sortOnOptions').click();
+    cy.findByText('Effective date').click();
+    //save page
```
Comment (stevepiercy) [inline packages/volto/cypress/tests/core/blocks/blocks-search.js:645]:
```suggestion
    // save page
```

File: packages/volto/cypress/tests/core/blocks/blocks-search.js [hunk 6]
```diff
@@ -558,4 +558,102 @@ describe('Search Block Tests', () => {
...
+    cy.get('#toolbar-save').click();
+    // then we are able to see label and sort option
+    cy.get('.sort-label').should('have.text', 'Sort on');
+    cy.get('#select-search-sort-on  ').click();
+    cy.findByText('Effective date').click({ force: true });
+    cy.get(
+      'div#select-search-sort-on.search-react-select-container.css-2b097c-container',
+    ).contains('Effective date');
+    cy.get('#select-search-sort-on  ').click();
```
Comment (stevepiercy) [inline packages/volto/cypress/tests/core/blocks/blocks-search.js:654]:
```suggestion
    cy.get('#select-search-sort-on').click();
```

File: packages/volto/cypress/tests/core/blocks/blocks-search.js [hunk 7]
```diff
@@ -558,4 +558,102 @@ describe('Search Block Tests', () => {
...
+    cy.findByText('Order in folder').click();
+    // Add one soron options below
+    cy.get('.field-wrapper-showSortOn .wrapper .ui label').click();
+    cy.get('#field-sortOnOptions').click();
+    cy.findByText('Effective date').click();
+    //save page
+    cy.get('#toolbar-save').click();
+    // then we are able to see label and sort option
+    cy.get('.sort-label').should('have.text', 'Sort on');
+    cy.get('#select-search-sort-on  ').click();
```
Comment (stevepiercy) [inline packages/volto/cypress/tests/core/blocks/blocks-search.js:649]:
```suggestion
    cy.get('#select-search-sort-on').click();
```

File: packages/volto/cypress/tests/core/blocks/blocks-search.js [hunk 8]
```diff
@@ -558,4 +558,102 @@ describe('Search Block Tests', () => {
...
+    cy.get('.field-wrapper-showSortOn .wrapper .ui label').click();
+    cy.get(
+      '#field-sortOnOptions > .react-select__control > .react-select__value-container ',
+    ).click();
+    cy.findByText('Effective date').click();
+    //save page
+    cy.get('#toolbar-save').click();
+    // then we are able to see label and sort option
+    cy.get('.sort-label').should('have.text', 'Sort on');
+    cy.get('#select-search-sort-on  ').click();
```
Comment (stevepiercy) [inline packages/volto/cypress/tests/core/blocks/blocks-search.js:611]:
```suggestion
    cy.get('#select-search-sort-on').click();
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
For `it` statements, we should use the common name, not a technical name. I removed trailing spaces from several selectors. I fixed some grammatical and other issues in comments and `it` statements.

File: packages/volto/cypress/tests/core/blocks/blocks-search.js (hunk 1 above)
Comment (iRohitSingh) [reply packages/volto/cypress/tests/core/blocks/blocks-search.js]: Done.

File: packages/volto/cypress/tests/core/blocks/blocks-search.js (hunk 4 above)
Comment (iRohitSingh) [reply packages/volto/cypress/tests/core/blocks/blocks-search.js]: Done.

File: packages/volto/cypress/tests/core/blocks/blocks-search.js (hunk 5 above)
Comment (iRohitSingh) [reply packages/volto/cypress/tests/core/blocks/blocks-search.js]: Done

File: packages/volto/cypress/tests/core/blocks/blocks-search.js (hunk 6 above)
Comment (iRohitSingh) [reply packages/volto/cypress/tests/core/blocks/blocks-search.js]: Done.

File: packages/volto/cypress/tests/core/blocks/blocks-search.js (hunk 8 above)
Comment (iRohitSingh) [reply packages/volto/cypress/tests/core/blocks/blocks-search.js]: Done.

File: packages/volto/cypress/tests/core/blocks/blocks-search.js (hunk 2 above)
Comment (iRohitSingh) [reply packages/volto/cypress/tests/core/blocks/blocks-search.js]: Done.

**Review by stevepiercy — APPROVED** [review]
LGTM.

**Comment by stevepiercy** [comment]
I'll rerun the failing CI checks to see if they're just flaky.

**Review by davisagli — APPROVED** [review]

**Pull request merged**